### ISSUE TITLE
newrelic_one_dashboard_json corrupts a dashboard if `variables` are used

### ISSUE THREAD

**Issue by potyl** [newrelic/terraform-provider-newrelic#2138]
### Terraform Version

This is happening with terraform via a github action, although it is not limited to that environment.

```
Run hashicorp/setup-terraform@v2
  with:
    terraform_version: 1.3.6

Initializing provider plugins...
- Finding newrelic/newrelic versions matching "3.8.0"...
- Installing newrelic/newrelic v3.8.0...
- Installed newrelic/newrelic v3.8.0 (signed by a HashiCorp partner, key ID DC9FC6B1FCE47986)
```

### Affected Resource(s)
- `newrelic_one_dashboard_json`

### Terraform Configuration

Terraform configuration:

```hcl
terraform {
  required_version = "~> 1.0"

  required_providers {
    newrelic = {
      source  = "newrelic/newrelic"
      version = "3.8.0"
    }
  }
}

provider "newrelic" {
  account_id = local.newrelic_account_id
  region     = "US"
}

resource "newrelic_one_dashboard_json" "variables" {
  json = file("dashboard-variables.json")
}
```

Dashboard json export:

```json
{
  "name": "Variable Terraform",
  "description": null,
  "permissions": "PUBLIC_READ_WRITE",
  "pages": [
    {
      "name": "Variable Terraform",
      "description": null,
      "widgets": [
        {
          "title": "",
          "layout": {
            "column": 1,
            "row": 1,
            "width": 4,
            "height": 3
          },
          "linkedEntityGuids": null,
          "visualization": {
            "id": "viz.billboard"
          },
          "rawConfiguration": {
            "facet": {
              "showOtherSeries": false
            },
            "nrqlQueries": [
              {
                "accountId": 264822,
                "query": "SELECT count(*) FROM Transaction"
              }
            ],
            "platformOptions": {
              "ignoreTimeRange": false
            }
          }
        }
      ]
    }
  ],
  "variables": [
    {
      "name": "name",
      "items": null,
      "defaultValues": [],
      "nrqlQuery": {
        "accountIds": [
          264822
        ],
        "query": "SELECT uniques(name) FROM Transaction"
      },
      "title": "Name",
      "type": "NRQL",
      "isMultiSelection": false,
      "replacementStrategy": "STRING"
    }
  ]
}
```

### Actual Behavior
If we create a dashboard with one variable (even if the variable is **not** used) and we export the dashboard via the JSON export button and we try to upload the dashboard with `newrelic_one_dashboard_json` the newly created dashboard is corrupt.

The UI shows the message:  _Oops! Something went wrong._

### Expected Behavior
The dashboard should be created without issues and any corruption.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Create a dashboard and add a variable with a simple NRQL query (e.g. `SELECT uniques(name) FROM Transaction`)
2. Export the dashboard from the web UI
3. Apply the dashboard with `newrelic_one_dashboard_json`

### Debug Output
[gist](https://gist.github.com/potyl/8904dce9689d9de1d1915fe57013c89c#file-newrelic_one_dashboard_json-log)

**Comment by roger-link** [comment]
I stumbled across something similar. However, when i apply i get an error like this:

`In field "variables": Expected type "[DashboardVariableInput!]", found [{defaultValue: {}, isMultiSelection: true, items: [{value: "value1"}, {value: "value2"}], name: "application", nrqlQuery: {accountIds: null, query: ""}, replacementStrategy: "STRING", title: "application", type: "ENUM"}].
│ In element #1: Expected type "DashboardVariableInput", found {defaultValue: {}, isMultiSelection: true, items: [{value: "value1"}, {value: "value2"}], name: "application", nrqlQuery: {accountIds: null, query: ""}, replacementStrategy: "STRING", title: "application", type: "ENUM"}.
│ In field "nrqlQuery": Expected type "DashboardVariableNrqlQueryInput", found {accountIds: null, query: ""}.
│ In field "accountIds": Expected type "[Int!]!", found null.`

My variables json looks like this(which was manually generated from ui):

```
  "variables": [
    {
      "name": "application",
      "items": [
        {
          "title": null,
          "value": "value1"
        },
        {
          "title": null,
          "value": "value2"
        }
      ],
      "defaultValues": [],
      "nrqlQuery": null,
      "title": "application",
      "type": "ENUM",
      "isMultiSelection": true,
      "replacementStrategy": "STRING"
    }
  ]
```

**Comment by gordonarogers** [comment]
We are seeing the exact same error as @roger-link is there a workaround or fix on the way for this?

**Comment by LucasLobo** [comment]
Same error on but using `newrelic_one_dashboard`, as reported here [2156](https://github.com/newrelic/terraform-provider-newrelic/issues/2156)

**Comment by kidk** [comment]
Thanks. This should be fixed in version [v3.10.0](https://github.com/newrelic/terraform-provider-newrelic/compare/v3.9.0...v3.10.0)
